erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -47,6 +48,17 @@ def validate_stop_addresses(self):
 			if not stop.customer_address:
 				stop.customer_address = get_address_display(frappe.get_doc("Address", stop.address).as_dict())
 
+	def is_visited(self):
+		for stop in self.delivery_stops:
+			if stop.visited:
```

Comment:
Check if the stop has a `delivery_note` too, otherwise this function will fail while trying to set the status